backend/src/main/java/ai/verta/modeldb/versioning/BlobDAORdbImpl.java

```diff
@@ -417,19 +417,19 @@ boolean childContains(Set<?> list, Set<?> sublist) {
       LOGGER.debug("Deleted location for Diff : {}", deletedLocations);
 
       // modified new blob location from the CommitA
-      Set<String> modifiedLocations = new LinkedHashSet<>(locationBlobsMapCommitB.keySet());
-      modifiedLocations.removeAll(addedLocations);
-      Map<String, BlobExpanded> commonBlobs =
-          locationBlobsMapCommitA.values().stream().collect(toMap(Entry::getValue, Entry::getKey));
-      commonBlobs
-          .keySet()
-          .retainAll(
-              locationBlobsMapCommitB.values().stream()
-                  .map(Entry::getValue)
-                  .collect(Collectors.toSet()));
-      Map<String, BlobExpanded> locationBlobsCommon =
-          getLocationWiseBlobExpandedMapFromCollection(commonBlobs.values());
-      modifiedLocations.removeAll(locationBlobsCommon.keySet());
+      Map<String, Set<BlobExpanded>> blobsB = getCollectToMap(locationBlobsMapCommitB);
+      Map<String, Set<BlobExpanded>> blobsA = getCollectToMap(locationBlobsMapCommitA);
+      for (Map.Entry<String, Set<BlobExpanded>> entry : blobsA.entrySet()) {
+        Set<BlobExpanded> ent = blobsB.get(entry.getKey());
```

Comment:
Commenting on how we compute the modified set will be helpful.

Reply:
fixed